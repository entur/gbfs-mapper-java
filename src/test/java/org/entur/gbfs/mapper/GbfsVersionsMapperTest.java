package org.entur.gbfs.mapper;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.annotations.SnapshotName;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import java.io.IOException;
import java.net.URL;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

@ExtendWith({SnapshotExtension.class})
class GbfsVersionsMapperTest {
    private static final ObjectMapper objectMapper = new ObjectMapper();

    private Expect expect;

    @SnapshotName("gbfs_v2_3_to_v3_0_gbfs_versions_file_snapshot")
    @Test
    void testMapGbfsVersionsFile() throws IOException {
        URL resource = getClass().getClassLoader().getResource("fixtures/v2_3/gbfs_versions.json");
        org.entur.gbfs.v2_3.gbfs_versions.GBFSGbfsVersions testSubject = objectMapper.readValue(resource, org.entur.gbfs.v2_3.gbfs_versions.GBFSGbfsVersions.class);
        org.entur.gbfs.v3_0.gbfs_versions.GBFSGbfsVersions mapped = GBFSMapper.INSTANCE.map(testSubject, "en");
        assertDoesNotThrow(() -> {
            expect
                    .serializer("json")
                    .toMatchSnapshot(mapped);
        });
    }

    @SnapshotName("gbfs_v3_0_to_v2_3_gbfs_versions_file_snapshot")
    @Test
    void testMapGbfsVersionsFileInverse() throws IOException {
        URL resource = getClass().getClassLoader().getResource("fixtures/v3_0/gbfs_versions.json");
        org.entur.gbfs.v3_0.gbfs_versions.GBFSGbfsVersions testSubject = objectMapper.readValue(resource, org.entur.gbfs.v3_0.gbfs_versions.GBFSGbfsVersions.class);
        org.entur.gbfs.v2_3.gbfs_versions.GBFSGbfsVersions mapped = GBFSMapper.INSTANCE.map(testSubject, "en");
        assertDoesNotThrow(() -> {
            expect
                    .serializer("json")
                    .toMatchSnapshot(mapped);
        });
    }
}
