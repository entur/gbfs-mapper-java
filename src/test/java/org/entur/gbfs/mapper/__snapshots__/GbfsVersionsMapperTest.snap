gbfs_v2_3_to_v3_0_gbfs_versions_file_snapshot=[
  {
    "last_updated": "2015-06-11T20:31:18.000+00:00",
    "ttl": 0,
    "version": "3.0-RC2",
    "data": {
      "versions": [
        {
          "version": "2.2",
          "url": "https://test.com/gbfs.json"
        }
      ]
    }
  }
]


gbfs_v3_0_to_v2_3_gbfs_versions_file_snapshot=[
  {
    "last_updated": 1434054678,
    "ttl": 0,
    "version": "2.3",
    "data": {
      "versions": [
        {
          "version": "2.2",
          "url": "https://test.com/gbfs.json"
        }
      ]
    }
  }
]